bad-bits/lib/badbits.js

```diff
@@ -0,0 +1,41 @@
+import { updateBadBitsDatabase } from './store.js'
+
+export const BADBITS_URL = 'https://badbits.dwebops.pub/badbits.deny'
+
+/**
+ * @param {Env} env
+ * @param {object} options
+ * @param {typeof globalThis.fetch} [options.fetchBadBits]
+ * @returns
+ */
+export async function fetchAndStoreBadBits(
+  env,
+  { fetchBadBits } = { fetchBadBits: globalThis.fetch },
+) {
+  const response = await fetchBadBits(BADBITS_URL)
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch badbits: ${response.status} ${response.statusText}`,
+    )
+  }
+
+  const text = await response.text()
+  const lines = text.split('\n')
+
+  const currentBadHashes = new Set()
+
+  for (const line of lines) {
+    if (line.trim() === '' || line.startsWith('# ')) {
+      continue
+    }
+
```

Comment:
```suggestion
```

Having this doesn't change the outcome of the calculation, because the condition below also handles these cases